relay-pii/src/regexes.rs

```diff
@@ -106,186 +106,239 @@ macro_rules! ip {
     (v6s) => { "[0-9a-fA-F]{1,4}" };
 }
 
+macro_rules! regex {
+    ($name:ident, $rule:expr) => {
+        #[allow(non_snake_case)]
+        mod $name {
+            use super::*;
+            pub static $name: Lazy<Regex> = Lazy::new(|| Regex::new($rule).unwrap());
+
+            #[test]
+            fn supports_byte_mode() {
+                assert!(regex::bytes::RegexBuilder::new($name.as_str())
+                    .unicode(false)
+                    .multi_line(false)
+                    .dot_matches_new_line(true)
+                    .build()
+                    .is_ok());
+            }
+        }
```

Comment:
This makes sure the regex can also be used by minidump scrubbing. There are several open questions (see below), but this was the most conservative approach I could think of.

1. Why does minidump scrubbing need `.unicode(false)`? https://github.com/getsentry/relay/blob/b5b738f4b2c9a546acaa67fb80026fe46bfadb41/relay-pii/src/attachments.rs#L29-L31
2. Could we automatically generate a non-unicode version of every regex, such that it does not have to be recompiled on the fly for every minidump?

Reply:
1. I read the issue and I still don't understand why this is necessary, would be nice if we had a test for this...
2. That'd be nice or if we didn't even need the special casing in the first place.